File: backend/src/App.Core/Entities/Account.cs
namespace App.Core.Entities;

/// <summary>
/// 会计科目实体（对应 PostgreSQL 表 Accounts）。
/// 科目树用 <see cref="ParentId"/> 自引用维护（<c>NULL</c> = 一级科目）；防环在 Handler 内沿父链上溯判定；
/// 「末级科目」（可记账）为派生规则（无子科目即末级），不落列
/// （specs/031-erp-finance-master/design.md §2.1）。
/// </summary>
public sealed class Account
{
    /// <summary>科目 ID</summary>
    public Guid Id { get; set; }

    /// <summary>科目编码，全局唯一（如 1001 / 100101）</summary>
    public string Code { get; set; } = string.Empty;

    /// <summary>科目名称</summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>科目类别（资产 / 负债 / 权益 / 成本 / 损益）</summary>
    public AccountCategory Category { get; set; }

    /// <summary>余额方向（借 / 贷）</summary>
    public AccountDirection Direction { get; set; }

    /// <summary>上级科目 id（<c>null</c> 表示一级科目）</summary>
    public Guid? ParentId { get; set; }

    /// <summary>同级排序（升序展示）</summary>
    public int SortOrder { get; set; }

    /// <summary>预置科目（系统种子写入，不可删除，可改名 / 停用）</summary>
    public bool IsPreset { get; set; }

    /// <summary>科目状态（启用 / 停用）</summary>
    public AccountStatus Status { get; set; } = AccountStatus.Enabled;

    /// <summary>备注，可空</summary>
    public string? Remark { get; set; }

    /// <summary>创建时间</summary>
    public DateTimeOffset CreatedAt { get; set; }

    /// <summary>更新时间</summary>
    public DateTimeOffset UpdatedAt { get; set; }

    /// <summary>创建人用户 id</summary>
    public Guid? CreatedBy { get; set; }

    /// <summary>更新人用户 id</summary>
    public Guid? UpdatedBy { get; set; }
}

File: frontend/src/api/account.ts
import { del, get, post, put } from './request'

/** 科目类别（1 资产 / 2 负债 / 3 权益 / 4 成本 / 5 损益） */
export type AccountCategory = 1 | 2 | 3 | 4 | 5

/** 余额方向（1 借 / 2 贷） */
export type AccountDirection = 1 | 2

/** 科目状态（0 停用 / 1 启用） */
export type AccountStatus = 0 | 1

/** 科目树节点（对应后端 AccountTreeNodeDto；children 已按同级排序，空集合表示末级科目） */
export interface AccountTreeNode {
  id: string
  code: string
  name: string
  category: AccountCategory
  direction: AccountDirection
  parentId: string | null
  sortOrder: number
  /** 预置科目（不可删除） */
  isPreset: boolean
  status: AccountStatus
  remark: string | null
  children: AccountTreeNode[]
}

/** 科目详情（对应后端 AccountDetailDto） */
export interface AccountDetail {
  id: string
  code: string
  name: string
  category: AccountCategory
  direction: AccountDirection
  parentId: string | null
  sortOrder: number
  /** 预置科目（不可删除） */
  isPreset: boolean
  status: AccountStatus
  remark: string | null
  createdAt: string
  updatedAt: string
}

export interface CreateAccountPayload {
  code: string
  name: string
  category: AccountCategory
  direction: AccountDirection
  parentId?: string
  sortOrder: number
  status: AccountStatus
  remark?: string
}

export interface UpdateAccountPayload {
  code: string
  name: string
  category: AccountCategory
  direction: AccountDirection
  parentId?: string
  sortOrder: number
  status: AccountStatus
  remark?: string
}

/** 查询会计科目树（全量，末级科目 = children 为空） */
export function getAccounts(): Promise<AccountTreeNode[]> {
  return get<AccountTreeNode[]>('/accounts')
}

/** 查询会计科目详情 */
export function getAccount(id: string): Promise<AccountDetail> {
  return get<AccountDetail>(`/accounts/${id}`)
}

/** 新增会计科目 */
export function createAccount(payload: CreateAccountPayload): Promise<AccountDetail> {
  return post<AccountDetail>('/accounts', payload)
}

/** 编辑会计科目（编码 / 名称 / 类别 / 方向 / 上级 / 排序 / 状态 / 备注全量覆盖；上级不得为自身或下级） */
export function updateAccount(id: string, payload: UpdateAccountPayload): Promise<AccountDetail> {
  return put<AccountDetail>(`/accounts/${id}`, payload)
}

/** 删除会计科目（预置科目、有子科目、已被凭证引用时后端拒绝） */
export function deleteAccount(id: string): Promise<null> {
  return del<null>(`/accounts/${id}`)
}

/** 停用 / 启用会计科目 */
export function updateAccountStatus(id: string, status: AccountStatus): Promise<AccountDetail> {
  return put<AccountDetail>(`/accounts/${id}/status`, { status })
}